app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/Variant/Attributes/ConfigurableAttributeIdV2.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ConfigurableProductGraphQl\Model\Resolver\Variant\Attributes;
+
+use Magento\Catalog\Model\Product;
+use Magento\Eav\Model\ResourceModel\Entity\Attribute;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * Format new option id_v2 in base64 encode for super attribute options
+ */
+class ConfigurableAttributeIdV2 implements ResolverInterface
+{
+    /**
+     * Option type name
+     */
+    private const OPTION_TYPE = 'configurable';
+
+    /**
+     * @var Attribute
+     */
+    private $eavAttribute;
+
+    /**
+     * @param Attribute $eavAttribute
+     */
+    public function __construct(Attribute $eavAttribute)
+    {
+        $this->eavAttribute = $eavAttribute;
+    }
+
+    /**
+     * Create a option id_v2 for super attribute in "<option-type>/<attribute-id>/<value-index>" format
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return string
+     *
+     * @throws GraphQlInputException
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['code']) || empty($value['code'])) {
+            throw new GraphQlInputException(__('Wrong format option data: "code" should not be empty.'));
+        }
+
+        $attributeId = $this->eavAttribute->getIdByCode(Product::ENTITY, $value['code']);
```

Comment:
Using eavAttribute adds a hard dependency. app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/Variant/Attributes.php:47 here the $value['options'] has the attribute_id which we need. Just make sure to return it along with the existing data that is being returned ('code' and other stuff) so you can leverage 'attribute_id' it on the ConfigurableAttributeIdV2 resolver. This will remove the hard dependency.

Reply:
That makes sense. Thank you for this one too, totally agree 👍.